### DISCUSSION TITLE
Lark should be "programming" like ANTLR or EBNF

### DISCUSSION THREAD

**Discussion by ThatXliner** [github-linguist/linguist#5231] [Classification] [answered]
Lark is classified as "data". Yet EBNF, PEG, ANTLTR, etc, are classified as "programming". Is there any reason for this?

**Comment by lildude** [comment, marked as answer]
> Yet EBNF, PEG, ANTLTR, etc, are classified as "programming". I

🤔 EBNF is _not_:

https://github.com/github/linguist/blob/fcda9cfe7295c1d51409c2de5f111b9847ff427b/lib/linguist/languages.yml#L1287-L1288

Linguist doesn't know about PEG, but I agree ANTLR is.

As for why Lark is considered "data": from the original PR: https://github.com/github/linguist/pull/5049 (courtesy of @Alhadis)

> This PR adds support for Lark grammar files, a variant of EBNF used by the Lark parsing library.

So it'll be because EBNF is considered data. Why?

Support was added in https://github.com/github/linguist/pull/3312 which doesn't provide much info but the commit https://github.com/github/linguist/pull/3312/commits/33899b9d6bbe81acdca6c6b81606c4af5e0b1f2f states:

> Extended Backus–Naur form ([EBNF][]) is a metalanguage used to specify language grammars.

... and the wiki page linked states:

> EBNF is used to make a formal description of a formal language such as a computer programming language.

Which kinda aligns with EBNF _not_ being a programming language itself.

As for ANTLR's determination: that's been in Linguist for well over 8 years ago in https://github.com/github/linguist/pull/697 and hasn't been changed since then. The PR doesn't provide any details so it's possible it was incorrectly classified as "programming" waaaaay back then and maybe should be "data" like EBNF and Lark. 

Of course all of this could be wrong. I don't know as I know nothing about any of these languages 😁

**Reply by ThatXliner** [reply]
Wait, so ANTLR's been classified as programming for a long time? If that's the case, then it's hard to change that now...

Technically they should all be data or a programming language (EBNF, etc included). But if ANTLR turns out to be like this for a long time, then I guess it should stay like that.
